### PR TITLE
add experiment with llama-agents

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: config/agent.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+control_plane:
```
Comment (marcusschiesser) [inline config/agent.yaml:1]: we still need this file?

File: agents-docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      - CONTROL_PLANE_ORCHESTRATOR_TYPE=pipeline
+    networks:
+      - ragapp-network
+    depends_on:
+      - message_queue
+
+  travel_agent:
+    build: .
+    ports:
+      - "8001:8001"
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:31]: the agents can all run on 8000 (as they have a host name we don't need port mapping)

File: agents-docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      - CONTROL_PLANE_HOST=control_plane
+      - CONTROL_PLANE_PORT=8000
+      - MESSAGE_QUEUE_URL=http://message_queue:8100
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - CONTROL_PLANE_MODEL=gpt-3.5-turbo
+      - CONTROL_PLANE_ORCHESTRATOR_TYPE=pipeline
+    networks:
+      - ragapp-network
+    depends_on:
+      - message_queue
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:26]: this two configs are good

File: agents-docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  travel_agent:
+    build: .
+    ports:
+      - "8001:8001"
+    volumes:
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:32]: how about mounting the needed config files using `configs`: https://docs.docker.com/compose/compose-file/08-configs/ ?

File: agents-docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      - ./config:/app/config
+    environment:
+      - AGENT_HOST=travel_agent
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:35]: can't we just use `hostname` (without configuring it)

File: agents-docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      - AGENT_PORT=8001
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:36]: don't we already have PORT?

File: agents-docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      - CONTROL_PLANE_URL=http://control_plane:8000
+      - MESSAGE_QUEUE_URL=http://message_queue:8100
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - ENVIRONMENT=dev
+      - AGENT_NAME=travel_agent
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:41]: why is this necessary?

File: agents-docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      - AGENT_DESCRIPTION=I can use DuckDuckGo search to find information about a country.
+      - AGENT_PROMPT=Ask me about a country and I will provide you with some information about it.
+      - SYSTEM_PROMPT=You're a helpful travel agent, your task is use DuckDuckGo search to find information about a country. You just need to return the information you find.
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:44]: that's confusing, we should only have one prompt

File: src/app/main.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        # system is configured - / points to chat UI
+        return FileResponse("static/index.html")
+    else:
+        # system is not configured - redirect to onboarding page
+        return RedirectResponse(url="/admin/#new")
+
+
+def init_app() -> FastAPI:
+    environment = os.getenv("ENVIRONMENT", "")
+    agent_name = os.getenv("AGENT_NAME", None)
```
Comment (marcusschiesser) [inline src/app/main.py:32]: how about just using `AGENT=true` instead?

File: agents-docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      - MESSAGE_QUEUE_PORT=8100
+    networks:
+      - ragapp-network
+
+  control_plane:
+    build: ./src/control_plane
+    ports:
+      - "8000:8000"
+    environment:
+      - CONTROL_PLANE_HOST=control_plane
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:17]: as this is the control plane, why do we need the `CONTROL_PLANE_` prefix?

File: agents-docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+      - CONTROL_PLANE_PORT=8000
+      - MESSAGE_QUEUE_URL=http://message_queue:8100
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+      - CONTROL_PLANE_MODEL=gpt-3.5-turbo
+      - CONTROL_PLANE_ORCHESTRATOR_TYPE=pipeline
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:22]: 1. pipeline also needs OPENAI_KEY?

2. if we use pipeline, then we should configure it here. e.g.:
```
PIPELINE=[travel_agent,research_agent]
```

File: agents-docker-compose.yml (hunk 10 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:17]: Because we have multiple services now so it's good to add prefix to the env variable, whether we deploy it separately or not.

File: agents-docker-compose.yml (hunk 11 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:22]: > pipeline also needs OPENAI_KEY?

I also have this question too, but it requires construct a vector store engine there which require embedding model, so i still let the openapi there to easier testing.

> if we use pipeline, then we should configure it here. e.g.:

Yes, i thought about this but we cannot do that now. The reason is we need construct the orchestrator to launch the control_plane where those agents need the control_plane booting first. That's why i just hard-code the pipeline chain for now. 
Another thing is it seems that we just need a service definition to init the pipeline orchestrator so if we have a share config for the service then we would able to do that.

File: agents-docker-compose.yml (hunk 2 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:31]: Each services (agents and control plane) need to know there host and port to contact with message queue, i guess it's using "push" pattern for now.

File: agents-docker-compose.yml (hunk 5 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:35]: Yes, i think we can have a code to get the current running host instead of specify it here. Let me try that.

File: agents-docker-compose.yml (hunk 6 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:36]: See my comment above about the port

File: agents-docker-compose.yml (hunk 7 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:41]: agent name is a config for a agent and it's using as consumer key i think

File: agents-docker-compose.yml (hunk 8 above)
Comment (leehuwuj) [reply agents-docker-compose.yml:44]: For pipeline orchestrator we just need the system_prompt, the others is for agentic orchestrator, but i still let it there :D

File: config/agent.yaml (hunk 1 above)
Comment (leehuwuj) [reply config/agent.yaml:1]: Ah, we don't use this now, let me remove this. But using a config file as metadata for the service also a good idea, we'll think about this later.

File: agents-docker-compose.yml (hunk 10 above)
Comment (marcusschiesser) [reply agents-docker-compose.yml:17]: i think that pattern makes sense if we have a shared .env file (which we don't have), but what are other use cases?

File: agents-docker-compose.yml (hunk 2 above)
Comment (marcusschiesser) [reply agents-docker-compose.yml:31]: what i mean is that inside of this network `travel_agent:8000` and `research_agent:8000` are unique, so no need to use  `travel_agent:8000` and `research_agent:8001`

File: agents-docker-compose.yml (hunk 6 above)
Comment (marcusschiesser) [reply agents-docker-compose.yml:36]: don't understand that. we have only one port per container, right?

File: agents-docker-compose.yml (hunk 7 above)
Comment (marcusschiesser) [reply agents-docker-compose.yml:41]: fallback to `HOST`?

File: agents-docker-compose.yml (hunk 11 above)
Comment (marcusschiesser) [reply agents-docker-compose.yml:22]: you can share the config using `configs`

but if this pipelineorchestrator is to difficult to set up, how about just using the agentorchestrator?

File: config/.env [hunk 12]
```diff
@@ -20,7 +20,7 @@ EMBEDDING_DIM=1536
 APP_HOST=0.0.0.0
 
 # The port to start the backend app.
-APP_PORT=8000
+# APP_PORT=8000
```
Comment (marcusschiesser) [inline config/.env:23]: why changes in config?

File: agents-docker-compose.yml [hunk 13]
```diff
@@ -0,0 +1,79 @@
+services:
```
Comment (marcusschiesser) [inline agents-docker-compose.yml:1]: looks good!

File: main.py [hunk 14]
```diff
@@ -1,78 +1,19 @@
+import os
```
Comment (marcusschiesser) [inline main.py:1]: move to src/app?

Comment (marcusschiesser) [reply main.py:1]: together with Dockerfile and poetry

**Comment by marcusschiesser** [comment]
closing as we will use another multi-agent approach
